60_networks.Rmd

```diff
@@ -0,0 +1,855 @@
+plot(spring_netprops,
+     repulsion = 0.98,
+     rmSingles = TRUE,
+     shortenLabels = "intelligent",
+     labelScale = FALSE,
+     nodeSize = "eigenvector",
+     nodeSizeSpread = 3,
+     nodeColor = "cluster", 
+     hubBorderCol = "darkgray",
+     cexNodes = 1.8,
+     edgeTranspHigh = 20,
+     title1 = "Network on genus level", 
+     showTitle = TRUE,
+     cexTitle = 2.3,
+     mar = c(1, 3, 4, 8))
+
+legend(0.7, 1.1, cex = 1.7, title = "estimated correlation:",
+       legend = c("+","-"), lty = 1, lwd = 3, col = c("#009900","red"), 
+       bty = "n", horiz = TRUE)
+```
+
+### Highlight data features
+
+We now color nodes according to their phylum. The sizes of the nodes are scaled by the clr-transformed abundances of the genera, since SPRING uses this transformation to account for compositionality.
+
+```{r network_plot_single_phylum, fig.width=15, fig.height=10}
+# Get phyla names
+taxtab <- as.data.frame(rowData(tse_genus))
+phyla <- gsub("p__", "", taxtab[, "phylum"])
+names(phyla) <- taxtab[, "genus"]
+
+# Assign low abundant phyla to a "Other" class
+low_abundant <- names(table(phyla)[table(phyla) <= 3])
+phyla[phyla %in% low_abundant] <- "Other"
+
+# Take a look at the table
+table(phyla)
+phyla <- as.factor(phyla)
+
+# Define phylum colors
+phylcol <- c("cyan", "blue3", "red", "lawngreen", "yellow", "deeppink", "orange")
+
+plot(spring_netprops,
+     repulsion = 0.98,
+     rmSingles = TRUE,
+     shortenLabels = "intelligent",
+     labelScale = FALSE,
+     nodeSize = "clr",
+     nodeColor = "feature", 
+     featVecCol = phyla, 
+     colorVec =  phylcol,
+     hubBorderCol = "darkgray",
+     cexNodes = 1.8,
+     edgeTranspHigh = 20,
+     title1 = "Network on genus level", 
+     showTitle = TRUE,
+     cexTitle = 2.3,
+     mar = c(1, 10, 4, 6))
+
+# Add legends
+legend(0.7, 1.1, cex = 1.7, title = "estimated correlation:",
+       legend = c("+","-"), lty = 1, lwd = 3, col = c("#009900","red"), 
+       bty = "n", horiz = TRUE)
+
+# Colors used in the legend should be equally transparent as in the plot
+phylcol_transp <- colToTransp(phylcol, 60)
+
+legend(-1.8, 1.1, cex = 1.7, pt.cex = 2.5, title = "Phylum:", 
+       legend=levels(phyla), col = phylcol_transp, bty = "n", pch = 16) 
+```
+
+**A few things to observe:**
+
+-   Genera belonging to the same phylum tend to cluster together, though not perfectly.
+-   Genera with a low total count play a rather unimportant role in the network, i.e., they have a low centrality.
+-   There is only one negative edge in the network. This edge is between two clusters, as expected when using the "signed" transformation.
+
+## Differential network analysis
+
+In addition to the count matrix, the data set also includes information on the subjects' geographical location, gender, age, and diet.
+
+An interesting question to explore now is whether the skin microbiome differs between people whose diets are different.
+
+```{r sample_data}
+table(colData(tse_genus)$Diet)
```

Comment:
Use altExp?

Also note that colData(tse)$Diet can be written more shortly just tse$Diet